var/spack/repos/builtin/packages/neovim/package.py

```diff
@@ -84,13 +84,12 @@ class Neovim(CMakePackage):
     )
 
     # depend on virtual, lua-luajit-openresty preferred
-    depends_on("lua-lang")
     depends_on("luajit", when="~no_luajit")
-    depends_on("lua@5.1:5.1.99", when="+no_luajit")
+    depends_on("lua-lang@5.1", when="+no_luajit")
```

Comment:
This is better, for sure.  Would it help to have a conflict to say that lua *can't* be in the dependencies when luajit is selected?  I'm not 100% sure if we need it.

Reply:
I understand your point about the conflict, but I wouldn't consider it a problem of `neovim` but more in general. For this reason, I think it would be probably a better idea to add that conflict to the `LuaPackage`, but I'm not sure that it will catch all use-cases (and how good the idea actually is).